### PR TITLE
MULE-9295: Remove VM transport usage from DB

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java [hunk 1]
```diff
@@ -65,37 +68,47 @@ public void commitsChanges() throws Exception
...
-        MuleMessage response = client.send("vm://testRollback", TEST_MESSAGE, null);
-
-        assertTrue(response.getPayload() instanceof NullPayload);
-
-        List<Map<String, String>> result = selectData("select * from PLANET where POSITION=4", getDefaultDataSource());
-        assertRecords(result, new Record(new Field("NAME", MARS.getName()), new Field("POSITION", 4)));
+        try
+        {
+            runTransactionalFlow("jdbcRollback", ACTION_ALWAYS_BEGIN, new XaTransactionFactory());
+            fail("Expected 'java.lang.IllegalStateException'");
```
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java]: Why is expecting IllegalStateException and then catches only ComponentException?

File: tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java [hunk 2]
```diff
@@ -267,8 +275,92 @@ protected MuleEvent runFlow(String flowName) throws Exception
     }
 
     /**
-     * Runs the given flow with a default event expecting a failure.
-     * Will fail if there's no failure running the flow.
+     * Runs the given flow with a default event inside a transaction
+     * 
+     * @param flowName the name of the flow to be executed
+     * @param action See {@link TransactionConfig} constants
+     * @param factory
```
Comment (pablokraan) [inline tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java]: Add missing param description

File: tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java [hunk 3]
```diff
@@ -267,8 +275,92 @@ protected MuleEvent runFlow(String flowName) throws Exception
...
+    {
+        try
+        {
+            this.runTransactionalFlow(flowName, payload, action, factory);
+        }
+        catch (MessagingException e)
+        {
+            return e;
+        }
+        catch (Exception e)
```
Comment (pablokraan) [inline tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java]: Flow is executed inside a ExecutionCallback#process which throws Exception. Why are you adding this restriction here? Is not better to expect an exception and then check that is of an expected type only when requested?

File: modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java [hunk 4]
```diff
@@ -65,37 +68,47 @@ public void commitsChanges() throws Exception
...
-        assertTrue(response.getPayload() instanceof NullPayload);
-
-        List<Map<String, String>> result = selectData("select * from PLANET where POSITION=4", getDefaultDataSource());
-        assertRecords(result, new Record(new Field("NAME", "Mercury"), new Field("POSITION", 4)));
+        try
+        {
+            runTransactionalFlow("rollbackWithNonTransactionalMP", ACTION_ALWAYS_BEGIN, new XaTransactionFactory());
+            fail("Expected 'java.lang.IllegalStateException'");
+        }
+        catch (ComponentException e)
```
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java]: Why can't you use runTransactionalFlowExpectingException on these test methods?

File: modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java (hunk 1 above)
Comment (elrodro83) [inline modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java]: The IllegalStateException is thrown by the test component, and will be wrapped by the Component exception. Will change the failure message to reflect that.

**Pull request merged**